src/VisualStudio/Core/Impl/CodeModel/Collections/ParameterCollection.cs

```diff
@@ -49,12 +50,13 @@ protected override bool TryGetItemByIndex(int index, out EnvDTE.CodeElement elem
 
         protected override bool TryGetItemByName(string name, out EnvDTE.CodeElement element)
         {
-            foreach (var parameter in this.ParentElement.GetParameters())
+            var parentNode = this.ParentElement.LookupNode();
+            if (parentNode != null)
             {
-                var childName = parameter.Name;
-                if (childName == name)
+                SyntaxNode parameterNode;
+                if (CodeModelService.TryGetParameterNode(parentNode, name, out parameterNode))
```

Comment:
I am just reusing an existing method here and we dont use the parameterNode because the information required is the nothing but the name which is already available